internal/controller/spinapp_controller.go

```diff
@@ -225,10 +226,37 @@ func (r *SpinAppReconciler) updateStatus(ctx context.Context, app *spinv1alpha1.
 	return nil
 }
 
+const defaultCASecretName = "spin-ca"
+
+// ensureDefaultCASecret creates the default ca certificate bundle in the
+// namespace of the app. Only one is required per namespace. The secret can be
+// overridden by the cluster operator.
+func (r *SpinAppReconciler) ensureDefaultCASecret(ctx context.Context, namespace string) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultCASecretName,
+			Namespace: namespace,
+		},
+		StringData: map[string]string{"ca-certificates.crt": cacerts.CACertificates()},
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: defaultCASecretName, Namespace: namespace}, secret)
+	if !apierrors.IsNotFound(err) { // secret is not not found
+		return nil
+	}
+
+	err = r.Client.Create(ctx, secret)
+	return client.IgnoreAlreadyExists(err)
+}
+
 // reconcileDeployment creates a deployment if one does not exist and reconciles it if it does.
 func (r *SpinAppReconciler) reconcileDeployment(ctx context.Context, app *spinv1alpha1.SpinApp, config *spinv1alpha1.ExecutorDeploymentConfig) error {
 	log := logging.FromContext(ctx).WithValues("deployment", app.Name)
 
+	if err := r.ensureDefaultCASecret(ctx, app.Namespace); err != nil {
```

Comment:
Strongly opposed to always doing this - across all namespaces - This should be a field on the SpinAppExecutor.

Something like:
```
installDefaultCACerts: bool
caCertSecret: string
// ^mutually exclusive
```

Reply:
Great idea

Reply:
I think it would be more intuitive if these weren't mutually exclusive. Meaning if both are set the operator would create the defaults secret as whatever name is set. If only `installDefaultCACerts` is set the operator would use the default name for the secret.

```
installDefaultCACerts: true
caCertSecret: myawesomesecret
```

Would create a secret with the default certs named `myawesomesecret`

The other option would be to validate that only one is set and return an error but that feels less friendly.

Any feels?

Reply:
hmm yeah that seems reasonable! - I think at that point we need to error somewhere if `installDefaultCACerts == true` and a secret already exists and wasn't created by SpinKube (to avoid having weird hidden issues)

Reply:
@adamreese that makes sense to me. If only `caCertSecret` is set, then do we assume that the user has already created that secret with a custom CA bundle?

Reply:
Yes. A user could also create/update a secret with the default name and it won't be deleted or modified.